scrambles/src/main/java/org/worldcubeassociation/tnoodle/puzzle/SquareOnePuzzle.java

```diff
@@ -1,40 +1,44 @@
 package org.worldcubeassociation.tnoodle.puzzle;
 
+import org.worldcubeassociation.tnoodle.scrambles.*;
 import org.worldcubeassociation.tnoodle.svglite.Color;
 import org.worldcubeassociation.tnoodle.svglite.Dimension;
 import org.worldcubeassociation.tnoodle.svglite.Svg;
 import org.worldcubeassociation.tnoodle.svglite.Transform;
 import org.worldcubeassociation.tnoodle.svglite.Path;
 import org.worldcubeassociation.tnoodle.svglite.Rectangle;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Iterator;
-import java.util.Random;
-
-import org.worldcubeassociation.tnoodle.scrambles.InvalidScrambleException;
-import org.worldcubeassociation.tnoodle.scrambles.Puzzle;
-import org.worldcubeassociation.tnoodle.scrambles.PuzzleStateAndGenerator;
+
+import java.util.*;
+
 import cs.sq12phase.FullCube;
 import cs.sq12phase.Search;
 import org.timepedia.exporter.client.Export;
 
 @Export
 public class SquareOnePuzzle extends Puzzle {
 
+    //public static final int SQ1_MAX_SCRAMBLE_LENGTH = 20;
+
     private static final int radius = 32;
 
+    private ThreadLocal<Search> twoPhaseSearcher;
+
     public SquareOnePuzzle() {
-        // TODO - we can't filter super aggresively until
-        // Chen Shuang's optimal solver is fixed.
-        //wcaMinScrambleDistance = 20;
         wcaMinScrambleDistance = 11;
+
+        twoPhaseSearcher = new ThreadLocal<Search>() {
+            protected Search initialValue() {
+                return new Search();
+            };
+        };
     }
 
     @Override
     public PuzzleStateAndGenerator generateRandomMoves(Random r) {
-        Search s = new Search();
-        String scramble = s.solution(FullCube.randomCube(r), Search.INVERSE_SOLUTION).trim();
+        FullCube randomState = FullCube.randomCube(r);
+
+        //String scramble = twoPhaseSearcher.get().solutionOpt(randomState, SQ1_MAX_SCRAMBLE_LENGTH, Search.INVERSE_SOLUTION).trim();
+        String scramble = twoPhaseSearcher.get().solution(randomState, Search.INVERSE_SOLUTION).trim();
```

Comment:
Unsure which version to prefer -- solution certainly runs a bit faster than solutionOpt, and having sub-optimal scrambler sequences seems fine to me.

Please add your input on which version you prefer.

Reply:
Having sub-optimal sequences doesn't seem like an issue to me.
Out of curiosity do you know how long the sub-optimal generally are?
Like under 15/16 moves?

Reply:
I do not have any statistics, but as it turns out we had been using the same sub-optimal scrambles in previous versions already (see left, code is identical except I refactored `randomState` into its own variable)